lib/src/public_ext.dart

```diff
@@ -45,28 +53,64 @@ extension TextTranslateExtension on Text {
           textWidthBasis: textWidthBasis);
 }
 
+/// Strings extension method for acces to [tr()] and [plural()]
+/// Example : 
+/// ```drat
+/// 'title'.tr()
+/// 'day'.plural(21)
+/// ```
 extension StringTranslateExtension on String {
+  /// {@macro tr}
   String tr(
           {List<String> args, Map<String, String> namedArgs, String gender}) =>
       ez.tr(this, args: args, namedArgs: namedArgs, gender: gender);
+
+  /// {@macro plural}
   String plural(num value, {NumberFormat format}) =>
       ez.plural(this, value, format: format);
 }
 
-//BuildContext extension method for acces to Easy Localization
+/// BuildContext extension method for acces to [locale], [supportedLocales], [fallbackLocale], [delegates] and [deleteSaveLocale()]
+///
+/// Example :
+///
+/// ```drat
+/// context.locale = Locale('en', 'US');
+/// print(context.locale.toString());
+///
+/// context.deleteSaveLocale();
+///
+/// print(context.supportedLocales); // output: [en_US, ar_DZ, de_DE, ru_RU]
+/// print(context.fallbackLocale);   // output: en_US
+/// ```
 extension BuildContextEasyLocalizationExtension on BuildContext {
+  /// retrieve current locale
   Locale get locale => EasyLocalization.of(this).locale;
-  set locale(Locale val) => EasyLocalization.of(this).locale = val;
 
+  /// set and change app locale
+  set locale(Locale val) => EasyLocalization.of(this).locale = val;
+  /// retrieve List of supported locales.
   List<Locale> get supportedLocales =>
       EasyLocalization.of(this).supportedLocales;
-
+  
+  /// retrieve the locale when the locale is not in `supportedLocales`
```

Comment:
Get locale if the locale is not in the list of supported locales in `supportedLocales`